packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1360,3 +1360,39 @@ class KeepAlive extends ParentDataWidget<KeepAliveParentDataMixin> {
     properties.add(DiagnosticsProperty<bool>('keepAlive', keepAlive));
   }
 }
+
+/// A sliver that constrains the cross axis extent of its sliver child.
+///
+/// The [SliverConstrainedCrossAxis] takes an [extent] parameter and uses it as
+/// the cross axis extent of the [SliverConstraints] passed to the sliver child.
+/// The widget ensures that the [extent] is a value between 0 and the parent's
+/// cross axis extent.
+///
+/// This is useful when you want to apply a custom cross-axis extent constraint
+/// to a sliver child.
+class SliverConstrainedCrossAxis extends SingleChildRenderObjectWidget {
+  /// Creates a sliver that constrains the cross axis extent of its sliver child.
+  ///
+  /// The [extent] parameters must not be null.
+  const SliverConstrainedCrossAxis({
+    super.key,
+    required this.extent,
+    Widget? sliver,
```

Comment:
In performLayout, it asserts that this is not null. Should this be non-nullable and required?

Reply:
I was thinking that but I noticed several other proxy slivers made is nullable e.g. SliverIgnorePointer, SliverOpacity, SliverPadding, etc. Is there a particular reason for this?

Reply:
It really depends. There is not a specific reason either way.

Currently though, this API contradicts itself. sliver is allowed to be null here, but then in performLayout there is `assert(child != null);`

If null should not be allowed, then this really should not be nullable.